frontend/test/metabase/scenarios/collections/permissions.cy.spec.js

```diff
@@ -285,6 +312,42 @@ describe("collection permissions", () => {
                   cy.location("pathname").should("eq", "/collection/root");
                   cy.findByText("Orders").should("not.exist");
                 });
+
+                describe("adding question to dashboard", () => {
+                  beforeEach(() => {
+                    popover()
+                      .findByText("Add to dashboard")
+                      .click();
+                  });
+
+                  it("should list collections user has `write` access to", () => {
+                    cy.get(".Modal")
+                      .findByText("Orders in a dashboard")
+                      .should("exist");
+                  });
+
+                  it("should list dashboards user has `write` access to when using search", () => {
+                    cy.get(".Modal").within(() => {
+                      cy.icon("search").click();
+                      cy.findByPlaceholderText("Search").type(
+                        "Orders in{Enter}",
+                      );
+                      cy.findByText("Orders in a dashboard").should("exist");
+                    });
+                  });
+
+                  it("should be able to add question to dashboard", () => {
+                    cy.get(".Modal")
+                      .as("modal")
+                      .findByText("Orders in a dashboard")
+                      .click();
+
+                    cy.get("@modal").should("not.exist");
+                    // By default, the dashboard contains one question
+                    // After we add a new one, we check there are two questions now
+                    cy.get(".DashCard").should("have.length", 2);
+                  });
+                });
```

Comment:
All of this can and should be a single test without an additional describe block. Again, the same suggestion applies as in the comment before - no need to explicitly use`.should("exist")`.

If you take a look at the test runner sidebar and the generated titles, you'll see that we don't have to explicitly say "user has `write` access to". Every member from `curate` group has write access to all collections.
![image](https://user-images.githubusercontent.com/31325167/115278034-422d1c00-a145-11eb-8d1e-a2d3adc7d4d7.png)


Also, I don't see any benefit in using/testing the search in this particular test because "Orders in a dashboard" was already shown in the modal. Testing search makes more sense for users who might not have write access to some collection because we want to make sure they are not even offered to them.

Reply:
Awesome! Merged the tests, removed explicit `"exists"` assertion, removed the "has write access" notes